### PR TITLE
Reset auth token if failed

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

This re-calls the function if the auth token failed

TODO:
- [ ] tests

## Checklist

- [ ] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [ ] **Prop changes**: [docs][docs] were updated
- [ ] **Prop changes**: E2E tests were updated, testing from the highest level possible

[docs]: https://ui.sandbox.manifold.co

### CODE REVIEW

File: src/utils/restFetch.ts [hunk 1]
```diff
@@ -68,9 +68,9 @@ export function createRestFetch({
 
     /* Handle expected errors */
     if (response.status === 401) {
-      setAuthToken('');
+      setAuthToken(null);
       report(response);
-      throw new Error('Auth token expired');
+      return restFetch(args); // retry
```
Comment (sslotsky) [inline src/utils/restFetch.ts]: Is there any logic in place to refresh the iframe when this happens? Or will that be in a separate PR?

Comment (dangodev) [reply src/utils/restFetch.ts]: Oh good call. I’m a bit stumped on how to go about doing that. What would you suggest?

Comment (dangodev) [reply src/utils/restFetch.ts]: You’re totally right—that’s the issue

Comment (sslotsky) [reply src/utils/restFetch.ts]: Been scratching my head on that one for a while. One possibility is to have shadowcat listen for the event that's getting reported when the 401 happens. But one potential difficulty there is that we could get a bunch of 401 errors at the same time, and reloading the iframe a bunch of times in a row is definitely not ideal.

Comment (dangodev) [reply src/utils/restFetch.ts]: Crazy idea: one old cachebusting trick is the ol’ arbitrary searchparam set by timestamp. e.g. `https://auth.manifold.co/oauth?t=123430938023`

We could do that, but it’d still have the problem of firing over and over again like you said.

We could clamp it to `5s` intervals or something, though and… yeah this is getting complicated 🤔

Comment (sslotsky) [reply src/utils/restFetch.ts]: I considered clamping to a time interval as well. It might not be awful.

Comment (dangodev) [reply src/utils/restFetch.ts]: Hm. Just tried dynamically creating an `iframe` in JS, and it sends one request every time you set the `src` even if you set it to the same thing over again. So we’d have to rely on Stencil’s `@Watch` decorator only allowing actual changes to come through (which I think it well)

File: src/components/manifold-auth-token/manifold-auth-token.tsx [hunk 2]
```diff
@@ -24,10 +24,20 @@ export class ManifoldAuthToken {
...
 
   setExternalToken(token?: string) {
-    if (token) {
-      if (!isExpired(token)) {
-        this.setAuthToken(token);
-      }
+    if (token && !isExpired(token)) {
+      // if token not expired, set
+      this.setAuthToken(token);
+    } else if (this.oauthUrl) {
```
Comment (sslotsky) [inline src/components/manifold-auth-token/manifold-auth-token.tsx]: I don't think clearing the auth token will result in the `token` prop being changed. That prop is being set by platforms. In order to achieve the implementation you're aiming at, you might have to inject the token from the tunnel as well, and set a new `@Watch()` on that property that kicks off this logic when it changes from present to null.

Comment (dangodev) [reply src/components/manifold-auth-token/manifold-auth-token.tsx]: Oh shoot. You’re right again.

Ugh my brain is breaking trying to follow all this stuff.

Comment (dangodev) [reply src/components/manifold-auth-token/manifold-auth-token.tsx]: It’s just frustrating because I’m looking at a Render dashboard right now that definitely doesn’t work after only a few minutes, and refreshing doesn’t help. I suspect it’s the localStorage stuff we added.

**Comment by dangodev** [comment]
Closing whatever was going on here… Best to start over again probably.
